src/app/dao/tests/PartDAO.spec.ts

```diff
@@ -60,43 +74,249 @@ describe('PartDAO', () => {
+        }
+    });
+    it('should allow verified users to read parts', async() => {
+        // Given a part and a verified user
+        await createConnectedGoogleUser('foo@bar.com', 'creator');
+        const partId: string = await partDAO.create(PartMocks.INITIAL);
+        await signOut();
+
+        await createConnectedGoogleUser('bar@bar.com', 'non-player');
+        // When reading the part
+        const result: Promise<MGPOptional<Part>> = partDAO.read(partId);
+        // Then it should succeed
+        await expectAsync(result).toBeResolvedTo(MGPOptional.of(PartMocks.INITIAL));
+    });
+    it('should forbid non-verified users to read parts', async() => {
+        // Given a part and a non-verified user
+        await createConnectedGoogleUser('foo@bar.com', 'creator');
+        const partId: string = await partDAO.create(PartMocks.INITIAL);
+        await signOut();
+
+        await createUnverifiedUser('bar@bar.com', 'non-verified');
+        // When reading the part
+        const result: Promise<MGPOptional<Part>> = partDAO.read(partId);
+        // Then it should fail
+        await expectFirebasePermissionDenied(result);
+    });
+    it('should allow owner to delete part', async() => {
+        // Given a part and its owner (as defined in the joiner)
+        const creatorUser: FireAuth.User = await createConnectedGoogleUser('foo@bar.com', 'creator');
+        const creator: MinimalUser = { id: creatorUser.uid, name: 'creator' };
+        const partId: string = await partDAO.create(PartMocks.INITIAL);
+        await joinerDAO.set(partId, { ...JoinerMocks.INITIAL, creator });
+
+        // When deleting the part
+        const result: Promise<void> = partDAO.delete(partId);
+
+        // Then it should succeed
+        await expectAsync(result).toBeResolvedTo();
+    });
+    it('should allow verified user to delete part if owner is not observing anymore', async() => {
+        // Given a part with its owner not observing this part
+        const creatorUser: FireAuth.User = await createConnectedGoogleUser('foo@bar.com', 'creator');
+        const creator: MinimalUser = { id: creatorUser.uid, name: 'creator' };
+        const partId: string = await partDAO.create(PartMocks.INITIAL);
+        await joinerDAO.set(partId, { ...JoinerMocks.INITIAL, creator });
+        // eslint-disable-next-line camelcase
+        const last_changed: Time = { seconds: Math.floor(Date.now() / 1000), nanoseconds: 0 };
+        await userDAO.update(creator.id, { observedPart: null, last_changed });
+        await signOut();
+
+        // and given another user
+        await createConnectedGoogleUser('bar@bar.com', 'visitor');
+
+        // When the other user deletes the part
+        const result: Promise<void> = partDAO.delete(partId);
+
+        // Then it should succeed
+        await expectAsync(result).toBeResolvedTo();
+    });
+    it('should allow verified user to delete part if owner has timed out', async() => {
```

Comment:
initial-part (non started, not ongoing, not finished)

Reply:
TODO